Row.c

```diff
@@ -468,27 +475,34 @@ void Row_printRate(RichString* str, double rate, bool coloring) {
    }
 
    if (!isNonnegative(rate)) {
-      RichString_appendAscii(str, shadowColor, "        N/A ");
-   } else if (rate < 0.005) {
-      int len = snprintf(buffer, sizeof(buffer), "%7.2f B/s ", rate);
-      RichString_appendnAscii(str, shadowColor, buffer, len);
+      RichString_appendAscii(str, shadowColor, "     N/A ");
    } else if (rate < ONE_K) {
-      int len = snprintf(buffer, sizeof(buffer), "%7.2f B/s ", rate);
+      int len = snprintf(buffer, sizeof(buffer), "%5.0fB/s ", rate);
       RichString_appendnAscii(str, baseColor, buffer, len);
    } else if (rate < ONE_M) {
-      int len = snprintf(buffer, sizeof(buffer), "%7.2f K/s ", rate / ONE_K);
+      double rateToPrint =  rate / ONE_K;
+      unsigned char precision = Row_getPrecision (rateToPrint);
+      int len = snprintf(buffer, sizeof(buffer), "%5.*fK/s ", precision, rateToPrint);
```

Comment:
1. `precision` takes the type `int`. You should set the data type properly.
2. There is a potential of converting this into a loop. You can see the code of `Meter_humanUnit()` function for how it can be done. There is a global `unitPrefixes` array that you can utilize.